everestrs/everestrs/src/everestrs_sys.cpp

```diff
@@ -69,37 +70,58 @@ inline ConfigField get_config_field(const std::string& _name, int _value) {
 
 } // namespace
 
+/// @brief The flag which prevents us from re-initializing the module.
+std::once_flag mod_flag;
+
+/// @brief The central handle to the EVerest infrastructure.
+std::unique_ptr<Module> mod;
```

Comment:
interface before was a little bit dangerous - this fixes it and makes sure that the module is init exactly once from the user code